Monika After Story/game/script-topics.rpy

```diff
@@ -1854,6 +1842,295 @@ label monika_rain_holdme:
         m 1dsc "Sorry..."
     return
 
+label monika_holdme_prep:
+    stop music fadeout 1.0
+
+    # clear selected track
+    $ songs.current_track = songs.FP_NO_SONG
+    $ songs.selected_track = songs.FP_NO_SONG
+    
+    # hide ui and disable hotkeys
+    $ HKBHideButtons()
+    $ store.songs.enabled = False
+    return
+
+label monika_holdme_start:
+    show monika 6dubsa
+    #Start the timer vv
+    $ start_time = datetime.datetime.now()
+
+    $ ui.add(PauseDisplayable())
+    $ ui.interact()
+
+    # renable ui and hotkeys
+    $ store.songs.enabled = True
+    $ HKBShowButtons()
+    return
+
+label monika_holdme_reactions:
+    if mas_isMoniLove():
+        if datetime.datetime.now() > (start_time + datetime.timedelta(minutes=30)):
```

Comment:
this works, but a cleaner notation is:
```python
if (datetime.datetime.now() - start_time) > datetime.timedelta(minutes=30):
```

however, since you need to do this for all the conditionals below, saving the diff at the start and comparing that is more efficient:
```python
elapsed_time = datetime.datetime.now() - start_time
if elapsed_time > datetime.timedelta(minutes=30):
   # stuff
elif elapsed_time > datetime.timedelta(minutes=10):
   # stuff
```